OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs

```diff
@@ -486,11 +480,18 @@ Action InitHotkeysPanel(Widget panel)
 					var added = new HashSet<HotkeyDefinition>();
 					var template = templates.Get(templateNode.Value.Value);
 					var remapDialogRoot = panel.Get("HOTKEY_DIALOG_ROOT");
-					var remapDialogPlaceholder = panel.GetOrNull("HOTKEY_DIALOG_PLACEHOLDER");
+
 					foreach (var t in types)
+					{
 						foreach (var hd in modData.Hotkeys.Definitions.Where(k => k.Types.Contains(t)))
+						{
 							if (added.Add(hd))
-								BindHotkeyPref(hd, modData.Hotkeys, template, hotkeyList, remapDialogRoot, remapDialogPlaceholder);
+							{
+								BindHotkeyPref(hd, modData.Hotkeys, template, hotkeyList, remapDialogRoot, isFirstHotkey);
+								isFirstHotkey = false;
```

Comment:
`isFirstHotkey` can the also hopefully go away (just set `selected` to the first hotkey here/above).

Reply:
I find it easier with a `isFirstHotkey` boolean. Otherwise the code has to check if it's the first element in each of the nested loops. I'm open to suggestions of course.